pulp_smash/pulp_smash_cli.py

```diff
@@ -0,0 +1,177 @@
+# coding=utf-8
+"""The entry point for Pulp Smash's user interface."""
+import json
+
+import click
+
+from pulp_smash import config, exceptions
+from pulp_smash.config import PulpSmashConfig
+
+
+def _raise_settings_not_found():
+    """Raise `click.ClickException` for settings file not found."""
+    result = click.ClickException(
+        'there is no settings file. Use `pulp-smash settings create` to '
+        'create one.'
+    )
+    result.exit_code = -1
+    raise result
+
+
+@click.group()
+def pulp_smash():
+    """Pulp Smash facilitates functional testing of Pulp."""
+
+
+@pulp_smash.group()
+@click.pass_context
+def settings(ctx):
+    """Manage settings file."""
+    cfg = PulpSmashConfig()
+    try:
+        cfg_path = cfg.get_config_file_path()
+    except exceptions.ConfigFileNotFoundError:
+        cfg_path = None
+    ctx.obj = {
+        'cfg_path': cfg_path,
+        'default_cfg_path': cfg.default_config_file_path,
+    }
+
+
+@settings.command('create')
+@click.pass_context
+def settings_create(ctx):
+    """Create a settings file."""
+    path = ctx.obj['cfg_path']
+    if path:
+        click.echo('Settings file already exist, continuing will override it.')
+        click.confirm('Do you want to continue?', abort=True)
+    else:
+        path = ctx.obj['default_cfg_path']
+    pulp_username = click.prompt('Pulp admin username', default='admin')
+    pulp_password = click.prompt('Pulp admin password', default='admin')
+    pulp_version = click.prompt('Pulp version')
+    system_hostname = click.prompt('System hostname')
+    if click.confirm('Is Pulp published using HTTP?'):
+        system_api_scheme = 'http'
+    else:
+        system_api_scheme = 'https'
+
+    if system_api_scheme == 'https' and click.confirm('Verify HTTPS?'):
+        certificate_path = click.prompt('SSL certificate path', default='')
+        if not certificate_path:
+            system_api_verify = True  # pragma: no cover
+        else:
+            system_api_verify = certificate_path
+    else:
+        system_api_verify = False
+
+    if click.confirm('Is Pulp\'s message broker Qpid?', default=True):
+        amqp_broker = 'qpidd'
+    else:
+        amqp_broker = 'rabbitmq'
+    using_ssh = not click.confirm(
+        'Are you running Pulp Smash on the Pulp system?')
+    if using_ssh:
+        click.echo(
+            'Pulp Smash will be configured to access the Pulp system using '
+            'SSH, because that, some additional information will be required.'
+        )
+        ssh_user = click.prompt('SSH username to use', default='root')
```

Comment:
I think it will be better to suggest some basic ssh config for the Pulp system and that is why the SSH username is being asked, but it won't do anything on the ssh config file. I still thinking if Pulp Smash should suggest that or should edit the ssh config or only mention that a passwordless connection should be able to the system.